File: examples/spaced/py/commands/alias.py
"""commands/alias.py — alias list/add/remove management commands."""
from __future__ import annotations

import os
import sys

import typer

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../../../sdk/py"))
from hop_top_kit.alias import load_from, save_to  # noqa: E402
from hop_top_kit.xdg import config_dir  # noqa: E402

app = typer.Typer(help="Manage command aliases.", add_completion=False)

_TOOL = "spaced"


def _config_path() -> str:
    return os.path.join(config_dir(_TOOL), "config.yaml")


@app.command("list")
def alias_list() -> None:
    """List all user-defined aliases."""
    try:
        aliases = load_from(_config_path())
    except FileNotFoundError:
        aliases = {}

    if not aliases:
        typer.echo("  No aliases defined.")
        return

    width = max(len(k) for k in aliases)
    for name in sorted(aliases):
        typer.echo(f"  {name:<{width}}  →  {aliases[name]}")


@app.command("add")
def alias_add(
    name: str = typer.Argument(..., help="Alias name"),
    target: str = typer.Argument(..., help="Target command expansion"),
) -> None:
    """Add or update an alias."""
    p = _config_path()
    try:
        aliases = load_from(p)
    except FileNotFoundError:
        aliases = {}

    aliases[name] = target
    save_to(p, aliases)
    typer.echo(f"  alias {name!r} → {target!r}")


@app.command("remove")
def alias_remove(
    name: str = typer.Argument(..., help="Alias name to remove"),
) -> None:
    """Remove an alias."""
    p = _config_path()
    try:
        aliases = load_from(p)
    except FileNotFoundError:
        aliases = {}

    if name not in aliases:
        typer.echo(f"  alias {name!r} not found", err=True)
        raise typer.Exit(1)

    del aliases[name]
    save_to(p, aliases)
    typer.echo(f"  removed alias {name!r}")

File: examples/spaced/ts/commands/alias.ts
/**
 * commands/alias.ts — alias management (list, add, remove)
 */

import { Command } from 'commander';
import * as path from 'path';
import { configDir } from '../../../../sdk/ts/src/xdg';
import { AliasStore } from '../../../../sdk/ts/src/alias';

function storePath(): string {
  return path.join(configDir('spaced'), 'config.yaml');
}

export function aliasCommand(): Command {
  const cmd = new Command('alias')
    .description('Manage command aliases');

  cmd
    .command('list')
    .description('List all aliases')
    .action(() => {
      const store = new AliasStore(storePath());
      store.load();
      const all = store.all();
      const names = Object.keys(all).sort();
      if (names.length === 0) {
        console.log('  No aliases configured.');
        return;
      }
      console.log('');
      for (const name of names) {
        console.log(`  ${name} → ${all[name]}`);
      }
      console.log('');
    });

  cmd
    .command('add <name> <target>')
    .description('Create an alias: name expands to target')
    .action((name: string, target: string) => {
      const store = new AliasStore(storePath());
      store.load();
      store.set(name, target);
      store.save();
      console.log(`  alias ${name} → ${target}`);
    });

  cmd
    .command('remove <name>')
    .description('Remove an alias')
    .action((name: string) => {
      const store = new AliasStore(storePath());
      store.load();
      if (!store.get(name)) {
        console.error(`  alias "${name}" not found`);
        process.exitCode = 1;
        return;
      }
      store.remove(name);
      store.save();
      console.log(`  removed alias "${name}"`);
    });

  return cmd;
}
